zipline/assets/_assets.pyx

```diff
@@ -298,6 +313,80 @@ cdef class Future(Asset):
         super_dict['contract_multiplier'] = self.contract_multiplier
         return super_dict
 
+"""
+TODO MB - rework this we need a Currency object and we need a FX pair
+
+ok we should use this a CurrencyPair, Currency could just be an enum?
```

Comment:
If we're extending 'Asset', and use an Asset.quote_currency field to define the quote currency for all Asset classes. EG:
```
eurusd = Currency(sid=1, base_currency='EUR', quote_currency='USD')
eurusd.quote_currency == 'USD'
eurusd.base_currency == 'EUR'
eurusd.symbol == 'EURUSD'

gbpusd = Currency(sid=2, base_currency='GBP', quote_currency='USD')
gbpusd.quote_currency == 'USD'
gbpusd.base_currency == 'GBP'
gbpusd.symbol == 'GBPUSD'

gbpeur = Currency(sid=3, base_currency='GBP', quote_currency='EUR')
gbpeur.quote_currency == 'EUR'
gbpeur.base_currency == 'GBP'
gbpeur.symbol == 'GBPEUR'
```

Even though sids 2 and 3 represent the same currency, it is quoted in two different currencies. This would also allow you to define which currency you want to use to pay for a new position in the base currency.

This would also mean that the Currency itself doesn't need to be represented as an asset, just a different type of cash maintained by the performance periods.